<?xml version="1.0" encoding="utf-8"?>	

<launch>

  <arg name="robot_name" default="ugv1" />
  <arg name="simulator" default="/vrep" />
  
  <arg name="respawn_value" default="false" /> <!-- boolean: true, false -->
  <arg name="use_marker_controller" default="true" /> <!-- boolean: true, false -->
  
  <arg name="enable_laser_proximity_callback" default="false"/>  <!-- enable a more conservative use of laser proximity info -->
  <arg name="lambda_trav" default="1.0"/>  

  <node name="path_planner_manager_$(arg robot_name)" pkg="path_planner" type="path_planner_manager" respawn="$(arg respawn_value)" output="screen"> 

        <param name = "robot_frame_name" value = "$(arg robot_name)/base_link"/>
        <param name = "int_marker_server_name" value = "$(arg simulator)/$(arg robot_name)/marker_controller"/>
        <param name = "int_marker_name" value = "$(arg robot_name)"/>
         
        <param name = "lambda_trav"  value="$(arg lambda_trav)"/>  
        
        <param name = "use_marker_controller" value = "$(arg use_marker_controller)"/>
        
        <param name = "path_planning_service_name" value = "$(arg simulator)/$(arg robot_name)/path_planning_service"/>
        
        <param name = "enable_laser_proximity_callback" value = "$(arg enable_laser_proximity_callback)"/>
            
        <!--param name = "goal_topic_name" value = "$(arg simulator)/$(arg robot_name)/goal_topic"/-->
        <!--param name = "goal_abort_topic_name" value = "$(arg simulator)/$(arg robot_name)/goal_abort_topic_name"/-->

        <remap from = "/clustered_pcl/wall" to = "$(arg simulator)/$(arg robot_name)/clustered_pcl/wall"/>
        <remap from = "/trav/traversability" to = "$(arg simulator)/$(arg robot_name)/trav/traversability"/> 
        
        <remap from = "/goal_topic" to = "$(arg simulator)/$(arg robot_name)/goal_topic"/>
        <remap from = "/goal_abort_topic" to = "$(arg simulator)/$(arg robot_name)/goal_abort_topic"/>
        <remap from = "/trajectory_control_abort_topic" to = "$(arg simulator)/$(arg robot_name)/trajectory_control_abort_topic"/>
        
        <remap from = "/robot_pp_path" to = "$(arg simulator)/$(arg robot_name)/robot_pp_path"/>

        <remap from = "/robot_path" to = "$(arg simulator)/$(arg robot_name)/robot_path"/>
        <remap from = "/robot_local_path" to = "$(arg simulator)/$(arg robot_name)/robot_local_path"/>
        
        <remap from = "/robot_path_draw" to = "$(arg simulator)/$(arg robot_name)/robot_path_draw"/>
        <remap from = "/robot_local_path_draw" to = "$(arg simulator)/$(arg robot_name)/robot_local_path_draw"/>

        <remap from = "/path_planner/visited_nodes" to = "$(arg simulator)/$(arg robot_name)/path_planner/visited_nodes"/>
        <remap from = "/path_planner/localPath" to = "$(arg simulator)/$(arg robot_name)/path_planner/localPath"/>
        
        <remap from = "/planner/tasks/path" to = "$(arg simulator)/$(arg robot_name)/planner/tasks/path"/>
        <remap from = "/planner/tasks/global_path" to = "$(arg simulator)/$(arg robot_name)/planner/tasks/global_path"/>
        <remap from = "/planner/tasks/feedback" to = "$(arg simulator)/$(arg robot_name)/planner/tasks/feedback"/>
        
        <remap from = "/path_planning_status" to = "$(arg simulator)/$(arg robot_name)/path_planning_status"/>
        
        <remap from = "/laser_proximity_topic" to = "$(arg simulator)/$(arg robot_name)/laser_proximity_topic"/>
    
  </node> 
  
</launch>
